hw/ip/otbn/data/insns.yml

```diff
@@ -0,0 +1,938 @@
+# fields:
+#
+#  name:      The name of the operand. Required and must be unique.
+#
+#  type:      The type of the operand. A string. If this can be figured out
+#             from the operand name, it is optional. See below for a list of
+#             possible operand types and the rules for recognising them
+#             automatically.
+#
+#  doc:       A fragment of markdown that documents the operand
+#
+# The valid operand types are as follows:
+#
+#  gpr:       The name of a general purpose register. Syntax "grs" for a
+#             source; "grd" for a destination.
+#
+#  wdr:       The name of a wide register. Syntax "wrs" for a source;
+#             "wrd" for a destination.
+#
+#  csr:       The name of a CSR. Syntax "csr" (always considered a destination)
+#
+#  wsr:       The name of a WSR. Syntax "wsr" (always considered a destination)
+#
+#  immediate: An immediate operand. Syntax "imm<n>" where n is the number of
+#             bits or just imm for an unspecified width.
+#
+#  enum:      An immediate with weird syntax. The syntax is "enum(a,b,c,d)"
+#             where a..d are the different possible syntaxes that can be used.
+#             The values are interpreted as immediates in enumeration order (so
+#             a is 0; d is 3).
+#
+#  option:    A 1-bit immediate with weird syntax. Written "option(foo)" to
+#             mean that if the string "foo" appears then the immediate has
+#             value 1 and if it doesn't the immediate has value 0.
+#
+# Operand types are inferred from names as follows:
+#
+#  grd:        GPR destination
+#  grs, grs<n> GPR source
+#  wrd:        WDR destination
+#  wrs, wrs<n> WDR source
+#  csr         CSR name
+#  wsr         WSR name
+#  imm, imm<n> Immediate
+#  offset      Immediate (unspecified width)
+
+insns:
+  - mnemonic: add
+    rv32i: true
+    synopsis: Add
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: addi
+    rv32i: true
+    synopsis: Add Immediate
+    operands: [grd, grs1, imm]
+
+  - mnemonic: lui
+    rv32i: true
+    synopsis: Load Upper Immediate
+    operands: [grd, imm]
+
+  - mnemonic: sub
+    rv32i: true
+    synopsis: Subtract
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: and
+    rv32i: true
+    synopsis: Bitwise AND
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: andi
+    rv32i: true
+    synopsis: Bitwise AND with Immediate
+    operands: [grd, grs1, imm]
+
+  - mnemonic: or
+    rv32i: true
+    synopsis: Bitwise OR
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: ori
+    rv32i: true
+    synopsis: Bitwise OR with Immediate
+    operands: [grd, grs1, imm]
+
+  - mnemonic: xor
+    rv32i: true
+    synopsis: Bitwise XOR
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: xori
+    rv32i: true
+    synopsis: Bitwise XOR with Immediate
+    operands: [grd, grs1, grs2]
+
+  - mnemonic: lw
+    rv32i: true
+    synopsis: Load Word
+    operands: [grd, offset, grs1]
+    syntax: <grd>, <offset>(<grs1>)
```

Comment:
Re the syntax: this is supposed to exactly match what was in the OTBN ISA docs before. I'm happy to change that, but not in this PR.

If there's an explicit syntax, the ordering of the operands list only affects the ordering of the table that lists them. In a later patch, I think I might enforce that this should match the ordering in the given syntax but, again, I wanted to match the existing documentation as much as possible for now because that makes eyeball-checking the results much easier.

The types etc. will eventually end up being inferred from encoding information. I'm working on that patch at the moment!